File: shared/db/schema/goals.ts
import { pgTable, uuid, varchar, text, boolean, jsonb } from "drizzle-orm/pg-core";

/**
 * GOALS Table
 * Defines hierarchical goals with attributes aligned with SQLAlchemy models.
 */
export const goals = pgTable("goals", {
  id: uuid("id").primaryKey().defaultRandom(),
  title: varchar("title", { length: 255 }).notNull(),
  description: text("description").notNull(),
  markdown_content: text("markdown_content").default(""),
  priority: varchar("priority", { length: 20 }), // critical, high, medium, low
  category: varchar("category", { length: 100 }),
  time_frame: varchar("time_frame", { length: 20 }), // weekly, monthly, yearly
  deadline: varchar("deadline", { length: 50 }),
  label_color: varchar("label_color", { length: 20 }).default("#ffffff"),
  assignee_initials: varchar("assignee_initials", { length: 10 }).default("--"),
  tags: jsonb("tags").default([]),
  links: jsonb("links").default([]),
  references: jsonb("references").default([]),
  internal_tasks: jsonb("internal_tasks").default([]),
  is_current_focus: boolean("is_current_focus").default(false),
});

File: backend/app/models/goals.py
from sqlalchemy import Column, String, Text, Boolean, JSON, ForeignKey, Integer
from sqlalchemy.orm import relationship
from .base import Base
import uuid

class Goal(Base):
    __tablename__ = "goals"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    title = Column(String, nullable=False)
    description = Column(Text, nullable=False)
    markdown_content = Column(Text, default="")
    priority = Column(String)  # critical, high, medium, low
    category = Column(String)
    time_frame = Column(String)
    deadline = Column(String, nullable=True)
    label_color = Column(String, default="#ffffff")
    assignee_initials = Column(String, default="--")
    tags = Column(JSON, default=list)
    links = Column(JSON, default=list)
    references = Column(JSON, default=list)
    internal_tasks = Column(JSON, default=list)
    is_current_focus = Column(Boolean, default=False)

    tasks = relationship("Task", back_populates="goal")

class Task(Base):
    __tablename__ = "tasks"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    title = Column(String, nullable=False)
    description = Column(Text, nullable=True)
    priority = Column(Integer, default=0)
    status = Column(String, default="Todo")
    tags = Column(JSON, default=list)
    deadline = Column(String, nullable=True)
    project_link = Column(String, nullable=True)
    is_archived = Column(Boolean, default=False)
    
    goal_id = Column(String, ForeignKey("goals.id"), nullable=True)
    goal = relationship("Goal", back_populates="tasks")
